src/backend/joanie/core/filters/admin/__init__.py

```diff
@@ -300,3 +300,47 @@ def filter_by_query(self, queryset, _name, value):
             | Q(owner__first_name__icontains=value)
             | Q(owner__last_name__icontains=value)
         ).distinct()
+
+
+class DiscountAdminFilterSet(filters.FilterSet):
+    """
+    DiscountAdminFilterSet allows to filter this resource with a query for amount and rate.
+    """
+
+    class Meta:
+        model = models.Discount
+        fields: List[str] = ["query"]
+
+    query = filters.CharFilter(method="filter_by_query")
+
+    def filter_by_query(self, queryset, _name, value):
+        """
+        Filter resource by looking for amount and rate which contains provided value in
+        "query" query parameter.
+        """
+
+        def amount_filter(amount, exact=False):
+            if exact and amount:
+                return Q(amount=amount)
+            return Q(amount__icontains=amount)
+
+        def rate_filter(rate, exact=False):
+            try:
+                rate = int(rate) / 100
+            except ValueError:
+                pass
+
+            if exact and rate:
+                return Q(rate=rate)
+            return Q(rate__icontains=rate)
+
+        if "€" in value:
+            value = value.replace("€", "")
```

Comment:
I just realized something weird... the discount model is not aware of the currency setup for the project. This is just a display concern but it could be harder and harder to change with time.

https://github.com/openfun/joanie/blob/9257d77e355c67a15d9550fab04a5f548a1bda31/src/backend/joanie/settings.py#L359-L360